lib/Frontend/CompilerInvocation.cpp

```diff
@@ -2107,8 +2107,11 @@ static bool ParseDiagnosticArgs(DiagnosticOptions &Opts, ArgList &Args,
       Args.hasFlag(OPT_color_diagnostics,
                    OPT_no_color_diagnostics,
                    /*Default=*/llvm::sys::Process::StandardErrHasColors());
-  // If no style options are specified, default to LLVM style.
-  Opts.PrintedFormattingStyle = DiagnosticOptions::FormattingStyle::Swift;
+  // If no style options are specified, default to Swift style, unless we are in
+  // caching mode which currently on supports LLVM style.
+  Opts.PrintedFormattingStyle = Args.hasArg(OPT_cache_compile_job)
```

Comment:
IIUC this is changing the default; if swift-style diagnostics are not supported with caching shouldn't we override the -diagnostic-style option if it's specified?